API.yaml

```diff
@@ -1829,6 +1881,11 @@ components:
           type: array
           items:
             type: string
+        labels:
+          description: Displayed labels of each column if tree has table columns. The length of the array and the index of each column correspond to the header array return in the tree model. Optional, tree has no columns if absent.
```

Comment:
What's the relation with the name? If we provide labels, do we still need the name?  Or should we replace the name by the labels?

Reply:
There should be also a name column which should match the name. In the Trace Compass server implementation it's index 0, but it doesn't have to be. Yes, it's redundant information. We could make the labels array mandatory and with at least length 1 for the name. What do you think?

Reply:
In terms of API, I think it would make more sense as there would be only one way of expressing the column's data. Otherwise, one would have the choice of either the name or a single label? or would need to pick a main column to put as a name.

Reply:
I just noticed that the name element is supposed to be an array of strings?? Is that right?

Reply:
Ok, makes sense. I'll have to update this patch then

Reply:
> I just noticed that the name element is supposed to be an array of strings?? Is that right?

Aha, I thought the array for the names had been done by Simon at one point. So, it's basically already what we've discussed (have an array for the column labels)

The variable name "name" doesn't reflect that it is an array and the description is not right. This needs to be updated and can be part of this PR.

Reply:
Could you update the name too? Either 1- remove it entirely, replacing it with labels, or 2- update the documentation or 3- make it a simple string to match the description. I'd personnally prefer 1 to make the protocol more straightforward (it's not really API yet right?)

Reply:
I think 1- is already done and 2- i.e. description needs to be updated. Also the labels are not optional and are required.